waspc/src/Wasp/Psl/Parser/Common.hs

```diff
@@ -13,66 +13,105 @@ module Wasp.Psl.Parser.Common
     float,
     integer,
     SourceCode,
+    Parser,
   )
 where
 
-import Text.Parsec
-  ( alphaNum,
-    char,
-    letter,
+import Data.Functor (void)
+import Data.Void (Void)
+import Text.Megaparsec
+  ( Parsec,
+    between,
+    empty,
+    many,
+    manyTill,
+    notFollowedBy,
+    sepBy,
+    sepBy1,
+    takeWhileP,
+    try,
+    (<?>),
     (<|>),
   )
-import Text.Parsec.Language (emptyDef)
-import Text.Parsec.String (Parser)
-import qualified Text.Parsec.Token as T
+import qualified Text.Megaparsec.Char as C
+import qualified Text.Megaparsec.Char.Lexer as L
 
-type SourceCode = String
+type Parser = Parsec Void SourceCode
 
-whiteSpace :: Parser ()
-whiteSpace = T.whiteSpace lexer
+type SourceCode = String
 
 reserved :: String -> Parser ()
-reserved = T.reserved lexer
+reserved name =
+  lexeme $
+    try $
+      do
+        _ <- C.string name
+        notFollowedBy identLetter <?> ("end of " ++ show name)
```

Comment:
I am not 100% sure redundant parenthesses, but this should work fine.

```suggestion
reserved name =
  lexeme $ try $ C.string name >> (notFollowedBy identLetter <?> ("end of " ++ show name))
```

Reply:
> What is "end of ", that is error message?

It's the name of what you're matching for, will get shown in error messages. It has an alias called `label` which I think better communicates what it does. The code is copied/adapted from Parsec which is where it gets the style from, which I guess is a bit weird.